src/components/Card/CardList.tsx

```diff
@@ -1,69 +1,75 @@
 import "./Card.scss";
 
-import React, { useState } from "react";
-import { Container } from "react-bootstrap";
+import React, { useEffect, useState } from "react";
+import { CardGroup } from "react-bootstrap";
 import { useParams } from "react-router-dom";
 
 import { logger } from "../../utils/logger";
 import { CardItem, CardItemData } from "./CardItem";
 
 export interface CardListProps {
   cardItemData: CardItemData[];
+  selectCard: (card: CardItemData) => void;
+  removeCard: (card: CardItemData) => void;
 }
 
-export function CardList({ cardItemData }: CardListProps) {
-  const [lastDirection, setLastDirection] = useState("");
+export function CardList({
+  cardItemData,
+  selectCard,
+  removeCard,
+}: CardListProps) {
   const { category } = useParams();
+  const [cards, setCards] = useState<CardItemData[]>([]);
 
   // If no category is supplied, use all cards.
   // Otherwise, use cards with a matching category
-  const cards =
-    category === undefined
-      ? cardItemData
-      : cardItemData.filter((item) => {
-          // Ensure that "record-keeping" will match "record-keeping"
-          // And that "financial-planning" will match "financial planning"
-          const categoryWithoutHyphens = category.replaceAll("-", " ");
-          const categoriesWithoutHyphens = item.categories.map((c) =>
-            c.replaceAll("-", " ")
-          );
+  useEffect(() => {
```

Comment:
`useEffect` is for anything that depends on an external change, in this case fetching external data ([React docs](https://beta.reactjs.org/reference/react/useEffect#fetching-data-with-effects)). Before the page first renders, it will run this and populate `cards`. Any time we change something in the dependency array (the last arg in `useEffect(function, [dependencies])`) it will rerun the effect to fetch new data.

I do think we should use a data fetching framework that means we don't need these `useEffect` calls, but I haven't used them before and don't want to complicate things so quickly. Remix has caught my eye since the `react-router-dom` folks are the ones who developed it. See [here](https://beta.reactjs.org/learn/start-a-new-react-project#building-with-a-full-featured-framework) for more fetching frameworks.

Reply:
Per the docs, `useEffect` runs **after the initial render**, not before, and because we are providing an empty dependency array, it will only run once. See, e.g., https://beta.reactjs.org/reference/react/useEffect#examples-dependencies (_"If your Effect truly doesn’t use any reactive values, it will only run after the initial render."_).

Consequently, IIUC, this change only changes the timing of when we populate the cards: before or after the initial rendering. 

Is there a particular reason to populate the cards after the initial rendering? Like in the example in the docs you referenced about fetching data, I can imagine doing that if the rendering affected the cards, but that's not the case here.

Or have I misunderstood `useEffect`?

Reply:
Ah sorry I see what you mean, and indeed I was wrong that it populates `cards` before the first render. The first render happens (without `cards` - it's set to `undefined`), then an async call is made to get `cards`, and finally it gets re-rendered after with `cards` put in. 

When there is a synchronous call as we've done before, the component gets blocked until it's ready. The user will see a blank or incomplete page until that point.

So `useEffect` means that we show a component and can put a `spinner` or some loading icon until the data is fetched asynchronously.

Reply:
Okay, fair enough. I don't think this change was required by the CSS or the quiz efforts, but the reason for including it seems sound.

Reply:
I'm applying a boy scout rule as far as possible - improving things as I go along. Will try to tone it down since it's causing unnecessary confusion.